src/error.rs

```diff
@@ -39,6 +42,35 @@ impl From<GolemError> for QuicksilverError {
     }
 }
 
+impl Display for QuicksilverError {
+    fn fmt(&self, fmt: &mut Formatter) -> FmtResult {
+        match self {
+            QuicksilverError::ImageError(e) => e.fmt(fmt),
+            QuicksilverError::IOError(e) => e.fmt(fmt),
+            QuicksilverError::GraphicsError(e) => write!(fmt, "{:?}", e),
+            QuicksilverError::SurfaceImageError => write!(
+                fmt,
+                "An image was passed to Surface with non-exclusive reference"
+            ),
+            QuicksilverError::NoSurfaceImageBound => write!(
+                fmt,
+                "A surface operation was attempted with no image bound to the surface"
+            ),
+        }
+    }
+}
+
+impl Error for QuicksilverError {
+    fn source(&self) -> Option<&(dyn Error + 'static)> {
+        match self {
+            QuicksilverError::ImageError(e) => Some(e),
+            QuicksilverError::IOError(e) => Some(e),
+            // GolemError is not Error :-(
```

Comment:
You could PR upstream for that too 😛 
(That is an oversight on my part)

Reply:
OK, will do. I usually want to limit the DFS through pull requests somewhere, though 😇